File: 백준/Silver/27972. 악보는 거들 뿐/악보는 거들 뿐.c
#include <stdio.h>
#define max(a, b) (((a)>(b))?(a):(b))

int N;
int arr[500001];

int main(){
    scanf("%d", &N);
	
	for(int i=0;i<N;i++)
		scanf("%d", &arr[i]);
	
	int inc=1, dec=1, ans=1;
	for(int i=0;i<N-1;i++) {
		if(arr[i]<arr[i+1]){
			dec=1, inc++;
			ans=max(ans,inc);
		}
		else if(arr[i]>arr[i+1]){
			inc=1, dec++;
			ans=max(ans,dec);
		}
	}
	
	printf("%d", ans);
}

File: 백준/Silver/27972. 악보는 거들 뿐/악보는 거들 뿐.rs
use std::io::{self, Read, Write};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut iter = input.split_whitespace();
    
    let n: usize = iter.next().unwrap().parse().unwrap();
    let mut arr = vec![0; n];
    for i in 0..n {
        arr[i] = iter.next().unwrap().parse().unwrap();
    }

    let mut ans: usize = 1;
    let mut inc: usize = 1;
    let mut dec: usize = 1;
    for i in 1..n {
        if arr[i] > arr[i-1] {
            inc += 1;
            dec = 1;
            ans = ans.max(inc);
        } else if arr[i] < arr[i-1] {
            dec += 1;
            inc = 1;
            ans = ans.max(dec);
        }
    }

    let mut out = io::BufWriter::new(io::stdout().lock());
    writeln!(out, "{}", ans).unwrap();
}